src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -920,6 +920,7 @@ bool GenericConfigurationManagerImpl<ImplClass>::_IsFullyProvisioned()
 #if CHIP_DEVICE_CONFIG_ENABLE_JUST_IN_TIME_PROVISIONING
         (!UseManufacturerCredentialsAsOperational() && _OperationalDeviceCredentialsProvisioned()) &&
 #endif
+        (mFlags.Has(Flags::kIsServiceProvisioned) && mFlags.Has(Flags::kOperationalDeviceCredentialsProvisioned)) &&
```

Comment:
Can someone check if this is the correct thing to do and won't break any other platform? cc @cecille @bzbarsky-apple

Reply:
Is this change still needed now that https://github.com/project-chip/connectedhomeip/pull/8454 has merged?

In any case, this does not look right because outside of TestConfigurationMgr it looks to me like `kIsServiceProvisioned` is never set.  And `kOperationalDeviceCredentialsProvisioned` looks like it's never set at all.

Reply:
@bzbarsky-apple , actually this change was added because in the case of On-Network commissioning `_IsFullyProvisioned` is returning true after getting credentials of AP, which causes code to break. It should return false as the node is not fully provisioned yet, so to determine if the node is fully provisioned or not, It should check here if operational device credentials are provisioned, is this understanding right?

Reply:
I would pretty much assume that any use of `IsFullyProvisioned` is broken....  Whatever code is meant to advertise `Commissionable Mode = 1` should presumably just check whether we have operational credentials directly.  That's what `Mdns::StartServer` ends up doing (by calling the kinda-broken `GetCurrentNodeId`; we should have a way to actually ask the boolean "do we have an opcred?" question we really want to ask.

_Does_ `kOperationalDeviceCredentialsProvisioned` ever get set?

Reply:
Yes, You are right.. `kOperationalDeviceCredentialsProvisioned` is never set

Reply:
Right.  So fundamentally:

1. `IsFullyProvisioned` in its current form can't really be relied on to ever become true; see https://github.com/project-chip/connectedhomeip/issues/7911 for another example, where any device with thread or wifi support that is actually connecting via ethernet will always have `IsFullyProvisioned` false.
2. We should probably not add new conditions to `IsFullyProvisioned`
3. Whatever code is misbehaving because it uses `IsFullyProvisioned` should probably be changed to test the things it actually cares about.